Framework/API/inc/MantidAPI/IFunction.h

```diff
@@ -194,6 +203,22 @@ class MANTID_API_DLL IFunction {
     virtual T apply(bool &) const = 0;
     /// Implement this mathod to access attribute as vector
     virtual T apply(std::vector<double> &) const = 0;
+
+    /// Evaluates the validator associated with attribute this visitor is to visit.
+    template <typename T1> void evaluateValidator(T1 &inputData) const {
```

Comment:
I thought it may be worth adding that the `evaluateValidator` function is essentially repeated in the code 3 times (once in each attribute visitor, once in the attribute itself). I have a further commit where I take the occurrences of this function and put them inside a class at `IFunction::validatorEvaluator`. Each attribute and visitors instantiate an object of this class upon construction to access its functions. I wasn't sure if I preferred this over the simpler but more repetitive version here, I'm open to opinions.

Reply:
I've tried to implement this, but can't get it working. I've created a toy project to check my implementation and in that, it works. I'm hoping a fresh pair of eyes can spot the issue. If you get a chance, would you mind taking a look:
[https://github.com/mantidproject/mantid/commit/0122bb4532d974455b5af23677f739a91eaf0771](url)